engine/order_manager.go

```diff
@@ -589,6 +589,14 @@ func (m *OrderManager) processSubmittedOrder(newOrderResp *order.SubmitResponse)
 		return nil, err
 	}
 
+	if err := m.orderStore.add(detail.CopyToPointer()); errors.Is(err, ErrOrdersAlreadyExists) {
+		// Streamed by ws before we got here. Details from ws supecede since they are more recent.
+		detail = m.orderStore.getByDetail(detail)
+	} else if err != nil {
+		// Non-fatal error. OrderStore doesn't mean it wasn't submitted and consumers shouldn't get an error
```

Comment:
> OrderStore doesn't mean it wasn't submitted

? I'm fine with the rest, but that isn't really clear to me

Reply:
I had several cases where Submitting an order through OrderManager resulted in a an error storing it, and that was all we got back.
The actual order was made. The response details had order details, including an orderID.
But instead we just returned nil and a formatted error and the consumer wasn't able to programatically recover, or anything.

The point here is to say that when orderManager Submits an order, storing it is secondary to success, and it will not error on failure to store.
The alternative would be to return the detail + the err, but it would be non-idiomatic Go to expect the consumer to catch err != nil and yet still use returned detail at all.

I tried both for a while and landed on this.

Reply:
What I was getting at was the comment wasn't clear 😄 Yes it can be strange to submit an order and get an error back about how it couldn't be stored which has little to do with the order itself, so logging it is preferable and can be tracked and sorted later.

Perhaps Something like:
`// Non-fatal error: Unable to store order, but error does not need to be returned to caller`

Reply:
Agreed. Done. 7f472996